@rtibbles 
for `test_full_import_with_newer_version` i am getting following error:

    def raise_(
        exception, with_traceback=None, replace_context=None, from_=False
    ):
        r"""implement "raise" with cause support.
    
            :param exception: exception to raise
            :param with_traceback: will call exception.with_traceback()
            :param replace_context: an as-yet-unsupported feature.  This is
             an exception object which we are "replacing", e.g., it's our
             "cause" but we don't want it printed.    Basically just what
             ``__suppress_context__`` does but we don't want to suppress
             the enclosing context, if any.  So for now we make it the
             cause.
            :param from\_: the cause.  this actually sets the cause and doesn't
             hope to hide it someday.
    
            """
        if with_traceback is not None:
            exception = exception.with_traceback(with_traceback)
    
        if from_ is not False:
            exception.__cause__ = from_
        elif replace_context is not None:
            # no good solution here, we would like to have the exception
            # have only the context of replace_context.__context__ so that the
            # intermediary exception does not change, but we can't figure
            # that out.
            exception.__cause__ = replace_context
    
        try:
           raise exception
E           sqlalchemy.exc.ArgumentError: SQL expression for WHERE/HAVING role expected, got <MagicMock name='Bridge().get_table().c.contentnode_id.in_()' id='140025039648992'>.

../../../.pyenv/versions/3.9.9/envs/kolibriNew/lib/python3.9/site-packages/sqlalchemy/util/compat.py:211: ArgumentError
============================================ 1 failed, 4 passed in 16.73 seconds ============================================

--- a/kolibri/core/content/test/test_channel_import.py
+++ b/kolibri/core/content/test/test_channel_import.py
@@ -965,36 +965,37 @@ class NoVersionv040ImportTestCase(NoVersionv020ImportTestCase):
 @patch("kolibri.core.content.utils.channel_import.Bridge")
 @patch("kolibri.core.content.utils.channel_import.logger")
 @patch("kolibri.core.content.utils.channel_import.select")
-class ChannelImportTestCase(TestCase):
+@patch("kolibri.core.content.utils.channel_import.apps")
+class ChannelImportTestCase(ContentImportTestBase, TransactionTestCase):
+    name = CONTENT_SCHEMA_VERSION
+    legacy_schema = None
+
     def setUp(self):
+        super(ChannelImportTestCase, self).setUp()
         self.channel_id = "6199dde695db4ee4ab392222d5af1e5c"
         self.channel_version = 2
         self.current_channel = None
 
-    def test_channel_already_exists(self, select_mock, logger_mock, BridgeMock):
-        self.current_channel = ChannelMetadata.objects.create(
-            id=self.channel_id,
-            name="existing_channel",
-            min_schema_version="1",
-            root_id=self.channel_id,
-        )
+    def tearDown(self):
+        return super().tearDown()
+
+    def test_channel_already_exists(
+        self, select_mock, logger_mock, apps_mock, BridgeMock
+    ):
+        self.current_channel = ChannelMetadata.objects.get(id=self.channel_id)
         self.current_channel.version = self.channel_version
-        ChannelMetadata.save(self.current_channel)
+        self.current_channel.save()
         self.channel_import = ChannelImport(self.channel_id, "")
         self.channel_import.channel_version = self.channel_version
         result = self.channel_import.check_and_delete_existing_channel()
         self.assertFalse(result)
 
-    def test_partial_import_no_deletion(self, select_mock, logger_mock, BridgeMock):
-        self.current_channel = ChannelMetadata.objects.create(
-            id=self.channel_id,
-            name="test",
-            min_schema_version="1",
-            root_id=self.channel_id,
-        )
+    def test_partial_import_no_deletion(
+        self, select_mock, logger_mock, apps_mock, BridgeMock
+    ):
+        self.current_channel = ChannelMetadata.objects.get(id=self.channel_id)
         self.current_channel.version = self.channel_version
-        ChannelMetadata.save(self.current_channel)
-        print("from", self.current_channel.version)
+        self.current_channel.save()
         self.channel_import = ChannelImport(self.channel_id, "")
         self.channel_import.channel_version = self.channel_version
         self.channel_import.partial = True
@@ -1002,38 +1003,32 @@ class ChannelImportTestCase(TestCase):
         result = self.channel_import.check_and_delete_existing_channel()
         self.assertTrue(result)
 
-    def test_partial_import_with_deletion(self, select_mock, logger_mock, BridgeMock):
+    def test_partial_import_with_deletion(
+        self, select_mock, logger_mock, apps_mock, BridgeMock
+    ):
         # Simulate partial import with the same version
-        self.current_channel = ChannelMetadata.objects.create(
-            id=self.channel_id,
-            name="existing_channel",
-            min_schema_version="1",
-            root_id=self.channel_id,
-        )
+        self.current_channel = ChannelMetadata.objects.get(id=self.channel_id)
         self.current_channel.version = self.channel_version
-        ChannelMetadata.save(self.current_channel)
+        self.current_channel.save()
         self.channel_import = ChannelImport(self.channel_id, "")
         self.channel_import.channel_version = self.channel_version - 1
         self.channel_import.partial = True
         result = self.channel_import.check_and_delete_existing_channel()
         self.assertFalse(result)
 
-    def test_full_import_with_newer_version(self, select_mock, logger_mock, BridgeMock):
+    def test_full_import_with_newer_version(
+        self, select_mock, logger_mock, apps_mock, BridgeMock
+    ):
         # Simulate full import with a newer version
-        self.current_channel = ChannelMetadata.objects.create(
-            id=self.channel_id,
-            name="existing_channel",
-            min_schema_version="1",
-            root_id=self.channel_id,
-        )
+        self.current_channel = ChannelMetadata.objects.get(id=self.channel_id)
         self.current_channel.version = self.channel_version
-        ChannelMetadata.save(self.current_channel)
+        self.current_channel.save()
         self.channel_import = ChannelImport(self.channel_id, "")
         self.channel_import.channel_version = self.channel_version + 1
         result = self.channel_import.check_and_delete_existing_channel()
-        self.assertFalse(result)
+        self.assertTrue(result)
 
-    def test_channel_not_exists(self, select_mock, logger_mock, BridgeMock):
+    def test_channel_not_exists(self, select_mock, logger_mock, apps_mock, BridgeMock):
         # Simulate channel not existing in the database
         self.channel_import = ChannelImport(self.channel_id, "")
         self.channel_import.channel_version = self.channel_version

